php_phongo.c

```diff
@@ -953,6 +952,13 @@ static bool php_phongo_apply_options_to_uri(mongoc_uri_t *uri, bson_t *options T
 			continue;
 		}
 
+        if (!strcasecmp(key, MONGOC_URI_READPREFERENCE)) {
```

Comment:
> You can start with any of these functions, but I think the general `php_phongo_apply_options_to_uri()` function may be best since you have an example already from my earlier comment.

Sorry for any miscommunication on my part. With that, I was suggesting that start with the options handled by that function (not read preference), since I had already provided a code example of how to validate int32 types.

> However I will fix this, but in this case we will pass the key literally in the exception message [readPreference], since we can not use key variable, and I think that it will not be proper to define it const char *key = bson_iter_key(&iter); for only that use

If you start with the three basic cases in `php_phongo_apply_options_to_uri()`, which are bool, int32, and utf8, then you have `key` available to use in the message already.

In the case of `php_phongo_apply_rp_options_to_uri()`, we know exactly what options to look and don't have to loop over all fields in the options document and fetch `key` at each position. While we could use a known constant like `MONGOC_URI_READPREFERENCE` in the exception message, I think `bson_iter_key(&iter);` may be preferable because it ensures that we report the exact case for the key of the invalid field.

Consider if the user had both "readPreference" and "readpreference" as keys in the options array.   `bson_iter_init_find_case()` will stop at the first one it finds, so `php_phongo_apply_rp_options_to_uri()` will only process the first field. I agree that this is an unfortunate edge case, but in this case I'd prefer to tell the user exactly which field we found to be invalid.